pkg/agent/openflow/network_policy.go

```diff
@@ -1248,15 +1261,19 @@ func (c *client) updateConjunctionMatchFlows(conj *policyRuleConjunction, newPri
 }
 
 // calculateFlowUpdates calculates the flow updates required for the priority re-assignments specified in the input map.
-func (c *client) calculateFlowUpdates(updates map[uint16]uint16) (addFlows, delFlows []binding.Flow, conjFlowUpdates map[uint32]flowUpdates) {
+func (c *client) calculateFlowUpdates(updates map[uint16]uint16, table binding.TableIDType) (addFlows, delFlows []binding.Flow,
+	conjFlowUpdates map[uint32]flowUpdates) {
 	conjFlowUpdates = map[uint32]flowUpdates{}
 	for original, newPriority := range updates {
 		originalPriorityStr := strconv.Itoa(int(original))
 		conjs, _ := c.policyCache.ByIndex(priorityIndex, originalPriorityStr)
-		klog.V(4).Infof("%d policyRuleConjunctions have flows installed at priority %v previously",
-			len(conjs), originalPriorityStr)
 		for _, conjObj := range conjs {
 			conj := conjObj.(*policyRuleConjunction)
+			// Only re-assign flow priorities for flows in the table specified.
+			if conj.ruleTableID != table {
+				klog.V(4).Infof("Conjunction %v with the same actionFlow priority is from a different table %v", conj.id, conj.ruleTableID)
```

Comment:
I remember conjunction ID is unique across tables, isn't it true any more?

Reply:
It is still true, but the filtered conjunctions might be from different tables. By filtering with the priorityIndex, the indexer will give policyRuleConjuctions of that priority from all tables, and for each such conjuction we need to check if the flow priorities need to be updated. This is kind of a workaround since go Indexer does not have something like `c.policyCache.ByIndex(priorityIndex, originalPriorityStr, tableIndex, tableID)`

Reply:
Got it